### PR TITLE
feat(nomos): Fix the JSON output

### LINKED ISSUE #953: The output of `nomossa -J -d /PATH/TO/FOLDER` is not valid JSON
If one calls `nomossa` with the `-J` option to produce JSON on an folder recursively, one gets non-valid JSON output as respose, e.g.
```
$ nomossa -J -d ./
["No_license_found"]["No_license_found"]["No_license_found"]["Zlib"]["No_license_found"]["See-doc.OTHER"]["No_license_found"]["No_license_found"]["Zlib-possibility"]["No_license_found"]["No_license_found"]["Zlib-possibility"]["Zlib"]["No_license_found"]["Zlib-possibility"]["No_license_found"]["No_license_found"][........]["No_license_found"]
```

### PR DESCRIPTION
**Pull request by GMishx**
## Description

Fix the JSON output while scanning using `-J` flag.

See old output in #953 

### Changes

The JSON output now following following syntax:

```json
{
  "results":[
    {
      "file": "relative-path-to-file.c",
      "licenses":[
        "EPL-1.0",
        "Apache-2.0"
      ]
    },
    {
      "file": "absolute-path-to-file-using-l-flag.c",
      "licenses":[
        "GPL-2.0",
        "LGPL-2.1"
      ]
    },
    ...
  ]
}
```

## How to test

1. Run nomos with `-J` flag on a single file
    - The result should be a valid JSON.
1. Run nomos with `-J -l` flags on a single file
    - The result should be a valid JSON with absolute file path.
1. Run nomos with `-J -d` flags on a directory
    - The result should be a valid JSON with results in an array.
1. Run nomos with `-J -l -d` flags on a directory
    - The result should be a valid JSON with absolute file path.
1. Run nomos without `-J` flag
    - The result should not be effected.
1. Run nomos from UI
    - The result should not be effected.

Closes #953

### CODE REVIEW

**Comment by maxhbr** [comment]
What was the old output format?

**Comment by GMishx** [comment]
> What was the old output format?

Please refer #953

File: src/nomos/mod_deps [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+          apt-get $YesOpt install libjson-c2;;
+        stretch|buster|sid|bionic|cosmic)
+          apt-get $YesOpt install libjson-c3;;
+        *) echo "ERROR: Unknown or Unsupported $DISTRO $CODENAME release, please report to the mailing list"; exit 1;;
+      esac;;
+    Fedora|RedHatEnterprise*|CentOS)
+      yum $YesOpt install \
+        json-c
+      ;;
+  esac
```
Comment (maxhbr) [inline src/nomos/mod_deps]: `*)` case for the `$DISTRO`?

File: src/nomos/mod_deps [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  case "$DISTRO" in
+    Debian|Ubuntu|LinuxMint)
+      apt-get $YesOpt install \
+        libjson-c-dev
+      ;;
+    Fedora|RedHatEnterprise*|CentOS)
+      yum $YesOpt install \
+        json-c-devel
+      ;;
+  esac
```
Comment (maxhbr) [inline src/nomos/mod_deps]: `*)` case for the `$DISTRO`?

File: src/nomos/mod_deps (hunk 1 above)
Comment (GMishx) [reply src/nomos/mod_deps]: Added.

File: src/nomos/mod_deps (hunk 2 above)
Comment (GMishx) [reply src/nomos/mod_deps]: Added.

**Comment by mcjaeger** [comment]
looks good to me (checking all switches), example:

```
vagrant@ubuntu-xenial:/usr/local/share/fossology/nomos/agent$ sudo ./nomos -J -l -d /home/vagrant/zlib-1.2.11
{
  "results":[
    {
      "file":"/home/vagrant/zlib-1.2.11/qnx/package.qpg",
      "licenses":[
        "No_license_found"
      ]
    },
    {
      "file":"/home/vagrant/zlib-1.2.11/zconf.h.cmakein",
      "licenses":[
        "Zlib-possibility"
      ]
    },
    {
      "file":"/home/vagrant/zlib-1.2.11/crc32.h",
      "licenses":[
        "No_license_found"
      ]
    },
    {
      "file":"/home/vagrant/zlib-1.2.11/zlib.h",
      "licenses":[
        "Zlib"
      ]
    },
...
```

**Comment by mcjaeger** [comment]
@maxhbr in the conversatin of this PR it looks resolved from your side, but the summary says "Changes Requested":

Screen Shot 2019-03-27 at 15 50 57

could you please check?

**Pull request merged**